src/parser/parser.c

```diff
@@ -5208,13 +5213,8 @@ static void parse_external_declaration(void)
 
 	position_t const *const pos = &ndeclaration->base.pos;
 	if (current_scope != file_scope) {
-		if (GNU_MODE) {
-			warningf(WARN_PEDANTIC, pos,
-			         "nested function %N is a GCC extension", ndeclaration);
-		} else {
-			errorf(pos,
-			       "nested function %N is a GCC extension", ndeclaration);
-		}
+		warningf(GNU_MODE? WARN_PEDANTIC: WARN_GNU_EXTENSION, pos,
+			"nested function %N is a GCC extension", ndeclaration);
```

Comment:
This changes the behaviour if GNU_MODE as well as WARN_GNU_EXTENSION are false.
Maybe you can use something like ```if (!warningf(WARN_GNU_EXTENSION, ...)) { warningf(WARN_PEDANTIC,...)}```